metaflow/plugins/argo/argo_inner_flow.py

```diff
@@ -1,31 +1,65 @@
 from metaflow.plugins.argo.argo_client import ArgoClient
 from metaflow.metaflow_config import KUBERNETES_NAMESPACE
+import time
 
 
 class TriggeredRun:
+    """
+    This class takes in information to identify and trigger a run of a
+    Metaflow flow using the Argo plugin. The run is triggered immediately 
+    upon initialization of this object, and the run is identified based on 
```

Comment:
Triggering an action at class construction is not they Pythonic way.

I recommend instead a `trigger_flow(flow_name, parameters, ...)` method that returns a `TriggeredRun` class on which one can query for status with `status, failed_steps,  exceptions`.

Reply:
done